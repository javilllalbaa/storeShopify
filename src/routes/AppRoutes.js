import React from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import { useEffect } from "react";
import { useSelector } from "react-redux";
import Header from "./../components/Header";
import ProductPage from "./../components/ProductPage";
import { selectData } from './../store/products'
import "./../App.scss";
import { useDispatch } from 'react-redux';
import { loadProduct } from './../store/products'

function AppRoutes() {

    const dispatch = useDispatch();

    let dataState = useSelector(selectData);
    let call =  false;
    let {
        product, 
        isFetchingProduct,
        selectedProduct
    } = dataState

    useEffect(() => {
        if(!call){
            dispatch(
                loadProduct()
            )
            call = true;
        }
    },[]);

    return (
        <Router>
            <Header product={product} />
            <Routes>
                <Route exact path="/" element={
                    <ProductPage product={product} isFetchingProduct={isFetchingProduct} selectedProduct={selectedProduct}  />
                }></Route>
            </Routes>
        </Router>
    )
}

export default AppRoutes