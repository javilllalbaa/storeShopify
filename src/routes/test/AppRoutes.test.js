import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

import AppRoutes from "../AppRoutes";

const testProps = {
    product: {
        id: "MLA9134324324",
        thumbnail: "image.png",
        title: "iPhone 11 64 Gb Blanco",
        price: { currency: "ARS", amount: "342343" },
        "attributes": [
            {
                "id": "BRAND",
                "name": "Marca",
                "value_id": "5627653",
                "value_name": "Vulcano BA",
                "value_struct": null,
                "values": [
                    {
                        "id": "5627653",
                        "name": "Vulcano BA",
                        "struct": null
                    }
                ],
                "attribute_group_id": "OTHERS",
                "attribute_group_name": "Otros"
            }
        ]
    },
    isFetchingProduct: false,
    selectedProduct: true
};

jest.mock('./../../store/products', () => ({
    selectData: jest.fn().mockReturnValue(testProps.product)
}));

jest.mock("react-redux", () => {
    return {
        useDispatch: () => jest.fn(),
        useSelector: () => jest.fn(),
        Provider: ({ children }) => children
    };
});

jest.mock("react-router-dom", () => {
    return {
        useLocation: jest.fn().mockReturnValue({ pathname: "/" }),
        useHistory: () => ({
            push: jest.fn(),
        }),
        BrowserRouter: () => jest.fn(),
        Router: () => jest.fn(),
        Route: () => jest.fn(),
        Switch: () => jest.fn(),
    };
});

describe("HomePage component", () => {
  describe("WHEN the component is render", () => {
    const wrapper = shallow(<AppRoutes />);

    it("THEN should display HomePage component", () => {
      expect(wrapper).toMatchSnapshot();
      wrapper.unmount()
    });

  });
});
