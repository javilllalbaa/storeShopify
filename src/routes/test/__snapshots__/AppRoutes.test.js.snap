// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component WHEN the component is render THEN should display HomePage component 1`] = `ShallowWrapper {}`;
